activitysim/abm/tables/disaggregate_accessibility.py

```diff
@@ -0,0 +1,267 @@
+    )
+    matched_df = choosers.reset_index().merge(matched_df, on=origin_col)
+    matched_df = matched_df.set_index(choosers.index.name)
+
+    return matched_df.loc[_idx]
+
+
+@inject.injectable()
+def disaggregate_suffixes():
+    return {"SUFFIX": None, "ROOTS": []}
+
+
+@inject.table()
+def maz_centroids():
+    df = input.read_input_table("maz_centroids")
+
+    if not df.index.is_monotonic_increasing:
+        df = df.sort_index()
+
+    logger.info("loaded maz_centroids %s" % (df.shape,))
+
+    # replace table function with dataframe
+    inject.add_table("maz_centroids", df)
+
+    return df
+
+
+@inject.table()
+def proto_disaggregate_accessibility():
+
+    # Read existing accessibilities, but is not required to enable model compatibility
+    df = input.read_input_table("proto_disaggregate_accessibility", required=False)
+
+    # If no df, return empty dataframe to skip this model
+    if not df:
+        return pd.DataFrame()
+
+    # Ensure canonical index order
+    if not df.index.is_monotonic_increasing:
+        df = df.sort_index()
+
+    logger.info("loaded proto_disaggregate_accessibility %s" % (df.shape,))
+
+    # replace table function with dataframe
+    inject.add_table("proto_disaggregate_accessibility", df)
+
+    return df
+
+
+@inject.table()
+def disaggregate_accessibility(persons, households, land_use, accessibility):
+    """
+    This step initializes pre-computed disaggregate accessibility and merges it onto the full synthetic population.
+    Function adds merged all disaggregate accessibility tables to the pipeline but returns nothing.
+
+    """
+
+    # If disaggregate_accessibilities do not exist in the pipeline, it will try loading csv of that name
+    proto_accessibility_df = pipeline.get_table("proto_disaggregate_accessibility")
+
+    # If there is no table, skip. We do this first to skip as fast as possible
+    if proto_accessibility_df.empty:
+        return pd.DataFrame()
+
+    # Get persons merged manually
+    persons_merged_df = inject.merge_tables(
+        persons.name, tables=[persons, households, land_use, accessibility]
+    )
+
+    # Extract model settings
+    model_settings = config.read_model_settings("disaggregate_accessibility.yaml")
+    merging_params = model_settings.get("MERGE_ON")
+    accessibility_cols = [
+        x for x in proto_accessibility_df.columns if "accessibility" in x
+    ]
+
+    # Parse the merging parameters
+    assert merging_params is not None
+
+    # Check if already assigned!
+    if set(accessibility_cols).intersection(persons_merged_df.columns) == set(
+        accessibility_cols
+    ):
+        return
+
+    # Find the nearest zone (spatially) with accessibilities calculated
+    # Note that from here on the 'home_zone_id' is the matched name
+    if "nearest_accessibility_zone_id" not in persons_merged_df.columns:
+        persons_merged_df = find_nearest_accessibility_zone(
+            persons_merged_df, proto_accessibility_df
+        )
```

Comment:
`method` arg is not passed into the function here. Users cannot define the method in settings. Judging from the code, `method` can be one of `{"centroid", "skims", ...}`. Is the implementation complete?

Reply:
Added user config functionality for this parameter. 

"# How the nearest sampled zone is found: skims or centroid distance
NEAREST_METHOD: skims"